client/signup/steps/site-type/index.jsx

```diff
@@ -115,6 +120,22 @@ export default connect(
 	( dispatch, ownProps ) => ( {
 		submitStep: ( siteTypeValue, themeRepo ) => {
 			dispatch( setSiteType( siteTypeValue ) );
+			dispatch(
+				recordTracksEvent( 'calypso_signup_actions_submit_site_type', {
+					value: siteTypeValue,
+				} )
+			);
+
+			let nextFlowName = ownProps.flowName;
+			if ( siteTypeValue === getSiteTypePropertyValue( 'id', 'store', 'slug' ) ) {
+				nextFlowName = 'ecommerce';
+			} else {
+				nextFlowName =
+					'ecommerce' === ownProps.flowName && ownProps.previousFlowName
+						? ownProps.previousFlowName
+						: ownProps.flowName;
```

Comment:
Can we reduce this to just `} else if ('ecommerce' === ownProps.flowName && ownProps.previousFlowName) { nextFlowName = ownProps. previousFlowName; }`?

Reply:
Thanks, fixed in [8ce58b6](https://github.com/Automattic/wp-calypso/pull/28672/commits/8ce58b6eea60717f8758c44edef851aaa3e13e38.)